src/app/components/auth/signup/RegisterForm.tsx

```diff
@@ -0,0 +1,370 @@
+    signupBtn,
+    nameLabel,
+    namePlaceholder,
+    emailLabel,
+    passwordLabel,
+    confirmPasswordLabel,
+    termsLabel,
+  } = locale[currentLang];
+
+  const [password, setPassword] = useState('');
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>({
+    mode: 'onChange',
+    resolver: yupResolver(formSchema),
+  });
+
+  const {
+    onChange: onNameChange,
+    onBlur: onNameBlur,
+    name: nameName,
+    ref: nameRef,
+  } = register('name');
+
+  const {
+    onChange: onEmailChange,
+    onBlur: onEmailBlur,
+    name: emailName,
+    ref: emailRef,
+  } = register('email');
+
+  const {
+    onChange: onPasswordChange,
+    onBlur: onPasswordBlur,
+    name: passwordName,
+    ref: passwordRef,
+  } = register('password');
+
+  const onChangePassword = async (e: ChangeEvent<HTMLInputElement>) => {
+    setPassword(e.target.value);
+    await onPasswordChange(e);
+  };
+
+  const {
+    onChange: onConfirmationChange,
+    onBlur: onConfirmationBlur,
+    name: confirmationName,
+    ref: confirmationRef,
+  } = register('passwordConfirmation');
+
+  const {
+    onChange: onTermsChange,
+    onBlur: onTermsBlur,
+    name: termsName,
+    ref: termsRef,
+  } = register('terms');
+
+  const [signUpError, setSignUpError] = useState<
+    SignUpErrorCodes | string | null
+  >(null);
+
+  const [type, setType] = useState('password');
+
+  const toggleType = () => {
+    if (type === 'password') {
+      setType('text');
+    } else {
+      setType('password');
+    }
+  };
+
+  async function handleSubmitEvent(data: FormData) {
+    const name = data.name;
+    const email = data.email;
+    const password = data.password;
+
+    try {
+      const response = await fetch('/api/auth/register', {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ name, email, password }),
+      });
+
+      if (response.status === 200) {
+        try {
+          const response = await fetch('/api/auth/', {
+            method: 'POST',
+            headers: {
+              Accept: 'application/json',
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(data),
+          });
+
+          if (response.status === 200) {
+            router.replace('/');
+          }
+        } catch (e) {
+          router.push('/auth/sign-in');
+        }
+      } else {
+        const data = (await response.json()) as ErrorResponse;
+        response.status === 401
+          ? setSignUpError(data.errorCode)
+          : setSignUpError(SignUpErrorCodes.UnknownError);
+      }
+    } catch (e) {
+      if (e instanceof Error) {
+        setSignUpError(SignUpErrorCodes.UnknownError);
+      }
+    }
+  }
+
+  return (
+    <form
+      className="space-y-8"
+      onSubmit={handleSubmit(handleSubmitEvent)}
+      noValidate
+    >
```

Comment:
There is form-provider from RHF which creates react-hook-form context. Take a look, that's more convenient

Reply:
@iamnkt
Please take this task.